### PR TITLE
Amber sarayu/client requests

### PR DESCRIPTION
**Pull request by AML441**
# Pull Request

## Description

Implemented table of client requests where users can click on entries and see the specific form each client filled out with corresponding search/sort features. View from admin and client side are different and match the Figma.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [X] I have performed a self-review and reviewed the changed files
- [X] My code follows the project's style guidelines and passes linting
- [X] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [ ] I have tested the changes manually, including edge cases

### Code Quality & Security
- [ ] I have formatted my code and removed debugging/unused code
- [X] No TypeScript errors or linting issues
- [ ] No sensitive information committed (API keys, passwords, etc.)
- [ ] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [ ] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)

image
image

## Additional Notes

TODO: Implement program field, apartment/unit field, and date field for client requests

## Summary by CodeRabbit

* **New Features**
  * Added Admin "Client Requests" page with filtering (All/New/Reviewed), search, sorting, and status management capabilities including approve/deny actions
  * Added Case Manager "Client Requests" page with search, filtering, and sorting options
  * Introduced detailed client request view displaying case manager, client, and furniture request information

### CODE REVIEW

File: app/client-requests/admin/page.tsx [hunk 1]
```diff
@@ -13,8 +52,137 @@ export default function ClientRequestsAdminPage() {
                     <SideNavbar />
                     <div className="flex-1 bg-[#F7F7F7] py-4 px-6 flex flex-col">
                         <h1 className="text-2xl text-primary font-extrabold">
-                            Client Requests (Admin)
+                            Client Requests
                         </h1>
+                        {/*top bar of options, need to filter the displayed info based on that*/}
```
Comment (joelchem) [inline app/client-requests/admin/page.tsx]: not gonna lie, i don't fully hate this pattern of having one page for all of this and using "changeGroup" to pick whats being filtered because it prevents duplicated code.

but i think just to maintain consistency with the other pages, have separate pages for New and Reviewed just like it is in the latest version of DonationRequests on main. I don't think you need the "All" page to be honest so dont include that for now.

this also lets you separate your protected route and tabs and stuff into the layout.tsx which keeps your functionality more segmented

File: components/client-requests/RequestDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { ClientRequest } from "@/types/client-requests";
+
+type ClientProps = {
+  client: ClientRequest;
+  userRole: string;
+};
+
+export function RequestDetailsPage({client, userRole} : ClientProps) {
```
Comment (joelchem) [inline components/client-requests/RequestDetails.tsx]: We can do a better job of matching the figma closely here i think. just small details like text size, text weight, padding, and stuff like that could definitely match better.

note: don't use px ofc! and also imo the nicest way to deal with text is like using "text-xs" "text-sm" etc. for text size. if you have the tailwind vscode plugin installed, as u type those, it'll tell you how many "px" each size corresponds to. For padding and stuff you can do "p-1", "p-2", "p-3", where each number represents 4px. This means "p-4" is the same as "p-[1em]" (bc 1em is 16px normally). the tailwind vscode plugin is v useful if you arent already using it! when you type in stuff like "p-3" it'll tell you in a popup how many "px" it corresponds to.

this was a lot of yap... lmk if it doesnt make sense lol

File: app/client-requests/case-manager/page.tsx [hunk 3]
```diff
@@ -16,6 +40,120 @@ export default function ClientRequestsCaseManagerPage() {
                         <h1 className="text-2xl text-primary font-extrabold">
                             Client Requests
                         </h1>
+
+                        <div className="flex gap-8 text-sm">
```
Comment (joelchem) [inline app/client-requests/case-manager/page.tsx]: Same thing about separate pages here that I mentioned above on the admin one

File: components/client-requests/RequestDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                                {/*TO DO: IMPLELMENT PROGRAM FIELD*/}
+                                <div className="p-2">{client.client.questions.hasMovedIn ? "Placeholder" : "Placeholder"} </div>
+                            </div>
+                        </div>
+                    </div>
+                    
+                    
+                )}
+            </div>
+            <div className="border-b border-gray-400 text-[#02AFC7] font-bold p-2 bg-[#FAFAFB]">Client</div>
```
Comment (joelchem) [inline components/client-requests/RequestDetails.tsx]: include the Name and Email as fields here for the Client... lowk if there's any info in the ClientRequest type it should probably be here in some capacity on this RequestDetails page. if u notice anything that's not displayed for some reason ask us about it please

**Review by joelchem — CHANGES_REQUESTED** [review]
Amazing work with this PR guys! There was lowk nothing huge to change for this one. The one slightly bigger structure change (the stuff with the pages you'll see in the review comments) is lowk only to be consistent with the rest of the app, and your current approach would otherwise be fine.

As always, let us know if you have any questions!

**Review by joelchem — APPROVED** [review]
